### ISSUE TITLE
FR: Add support for local files and media servers like Subsonic

### ISSUE THREAD

**Issue by isle9** [staniel359/muffon#102]

**Comment by staniel359** [comment]
@isle9 Hello.

Local files support is suggested in #73.
Could you describe this feature in details?

And what should Subsonic support look like?
I personally don't have any experience with it.

**Comment by fucksophie** [comment]
Hello, I have tons of experience with Subsonic.

Subsonic API is quite simple. 
Docs: http://www.subsonic.org/pages/api.jsp 

Basically, every single API request is authenicated by a MD5 salt + hash, and after that everything else is quite easy to implement.

TS implementation of the subsonic API I use in my music player (that's entierly Subsonic based): https://github.com/yourfriendoss/musicplayer2/blob/main/src/lib/subsonic.ts

**Comment by staniel359** [comment]
@yourfriendoss Hello.
So basically you create your own local media server and can connect to it via API?

**Comment by fucksophie** [comment]
> @yourfriendoss Hello. So basically you create your own local media server and can connect to it via API?

@staniel359 Yes! There are **many** servers and they all roughly support the same API with slight version differences, so you can create a local server on your laptop or your NAS for example, and just play music from your phone, PC, laptop, whatnot else.

Client implementations
- https://github.com/BLeeEZ/amperfy
- https://github.com/jeffvli/sonixd

Server implementations:
- https://github.com/sentriz/gonic
- https://airsonic.github.io/
- https://www.navidrome.org/ (not actually based on Subsonic - however has compat with latest subsonic version)

Subsonic API currently has some disagreements, but if you can create a API library based on the latest subsonic version I don't think it'll change much.

Important discussion: https://support.symfonium.app/t/subsonic-servers-participation/1233

**Comment by staniel359** [comment]
@yourfriendoss Thank you.
Okay, I'll work on it.

**Comment by charliefrance** [comment]
Would be great to have local file support as I use muffon to integrate all my sources of music. Some tracks I have on mp3 locally and no where else.
